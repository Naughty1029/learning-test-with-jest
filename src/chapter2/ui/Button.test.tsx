/**
* @jest-environment jsdom
*/
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react";
import Button from "./Button";


describe("Button", () => {
  test('renders button in the document', () => {
    const button = render(<Button />)
    fireEvent.click(button.getByText('ON'))
    expect(button.getByText(/OFF/i)).toBeTruthy()
    fireEvent.click(button.getByText('OFF'))
    expect(button.getByText(/ON/i)).toBeTruthy()
  })
})