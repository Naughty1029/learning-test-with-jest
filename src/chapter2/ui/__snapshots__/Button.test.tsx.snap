// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button in the document 1`] = `
<button
  className="primary"
  onClick={[Function]}
>
  ON
</button>
`;
